framework/test/driver/ResourceCompilerTest.cpp

```diff
@@ -422,28 +536,569 @@ static void testrcFailureModes(const std::string& rcbinpath, const std::string&
+        "foo" : [1, 2, 5, 7],
+        "bar" : "baz"
+    },
+    "embedded" : { 
+        "object" : { 
+            "array" : ["one", "two", "twenty"],
+            "string" : "boo",
+            "integer" : 42,
+            "float" : 101.1,
+            "boolean" : false
+        } 
+    },
+    "name" : " "
+    })";
+
+  createManifestFile(tempdir, manifest_json_part_1, "manifest_part1.json");
+  createManifestFile(tempdir, manifest_json_part_2, "manifest_part2.json");
+  createManifestFile(tempdir, manifest_json_part_3, "manifest_part3.json");
+
+  const std::string manifest_json = R"({
+    "bundle.symbolic_name" : "main",
+    "bundle.version" : "0.1.0",
+    "bundle.activator" : true,
+    "bundle.description" : "second manifest part",
+    "test" : { 
+        "foo" : [1, 2, 5, 7],
+        "bar" : "baz"
+    },
+    "embedded" : { 
+        "object" : { 
+            "array" : ["one", "two", "twenty"],
+            "string" : "boo",
+            "integer" : 42,
+            "float" : 101.1,
+            "boolean" : false
+        } 
+    },
+    "name" : " "
+    })";
+
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name " << "main";
+  cmd << " --out-file " << tempdir << "merged_zip.zip";
+  cmd << " --manifest-add " << tempdir << "manifest_part1.json";
+  cmd << " --manifest-add " << tempdir << "manifest_part2.json";
+  cmd << " --manifest-add " << tempdir << "manifest_part3.json";
+  US_TEST_CONDITION(EXIT_SUCCESS == runExecutable(cmd.str()), "Test the successful concatenation of multiple manifest.json files into one.");
+  
+  Json::Reader reader;
+  Json::Value root;
+  bool ok = reader.parse(manifest_json, root, false);
+
+  US_TEST_CONDITION(true == ok, "Test that the expected JSON content was parsed correctly.");
+
+  std::string expectedJSON(root.toStyledString());
+  // retrieve the JSON which was concatenated by usResourceCompiler
+  std::string concatenatedJSON;
+  US_TEST_NO_EXCEPTION_REQUIRED(concatenatedJSON = manifestContent(tempdir + "merged_zip.zip", "main"));
+
+  std::cout << "Expected JSON:\n\n" << expectedJSON << "JSON concatenated by usResourceCompiler:\n\n" << concatenatedJSON << std::endl;
+
+  // line feed and new line characters may be lurking in the strings. I don't know how to use miniz and jsoncpp to embed
+  // these characters in a consistent manner, so I'm opting to remove them afterwards.
+  concatenatedJSON.erase(std::remove(concatenatedJSON.begin(), concatenatedJSON.end(), '\r'), concatenatedJSON.end());
+  concatenatedJSON.erase(std::remove(concatenatedJSON.begin(), concatenatedJSON.end(), '\n'), concatenatedJSON.end());
+  expectedJSON.erase(std::remove(expectedJSON.begin(), expectedJSON.end(), '\r'), expectedJSON.end());
+  expectedJSON.erase(std::remove(expectedJSON.begin(), expectedJSON.end(), '\n'), expectedJSON.end());
+  
+  US_TEST_CONDITION(0 == concatenatedJSON.compare(expectedJSON), "Test that the concatenated JSON content matches the expected JSON content.");
```

Comment:
maybe `concatenedJSON == expectedJSON` ? :)

Reply:
They are functionally the same (operator== and std::string::compare). Is there a strong case for one over the other?